terraform/modules/azure-jenkinsinfra-storage/variables.tf

```
@@ -0,0 +1,28 @@
+# Required variables
+variable "service_fqdn" {
+  type = string
+}
+
+variable "storage_service_principal_ids" {
+  type = list(string)
+}
+
+variable "storage_service_principal_end_date" {
+  type = string
+}
+
+variable "storage_file_share_ids" {
+  type    = list(string)
+  default = []
+}
```

```suggestion
variable "storage_file_share_id" {
  type    = string
}
```

- Singular form (not sure to see the case for multiple filestorages: I would use different SPs instead) but not sure about this
- Whatever is the outcome of the above point, this variable should be mandatory (e.g. no default)

---

I've reproduced the plural form used for packer RG ids in https://github.com/jenkins-infra/shared-tools/blob/23ec7072579d46feca3893b262c5bef7bd91acd8/terraform/modules/azure-jenkinsinfra-controller/variables.tf#L100-L103

Singular form is fine to me.